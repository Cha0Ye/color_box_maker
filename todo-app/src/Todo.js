import React, { Component } from 'react';
// import './App.css';


class Todo extends Component {
    render() {
        return (
            <div className="Todo">
                { this.props.text }
                <button onClick={ this.props.triggerDelete }>X</button>
            </div>
        );
    }
}

export default Todo;
