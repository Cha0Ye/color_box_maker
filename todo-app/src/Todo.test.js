import React from 'react';
import Todo from './Todo';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

let wrapper;

beforeEach(() => {
    wrapper = mount(<Todo />);
})

//smoke test
it('renders without crashing', () => {
  const wrapper = shallow(<Todo />);
});

//snapshot test
it('matches snapshot', () => {

    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
});

//delete runs a function
it("runs a mocked fn on delete", function () {
    const deleteFn = jest.fn();
    let wrapper = mount(
      <Todo triggerDelete={deleteFn} />
    );
    const deleteBtn = wrapper.find("button").first();
  
    deleteBtn.simulate("click")
  
    expect(deleteFn).toHaveBeenCalled();
  });

  //for edit todo

it("adds items", function() {
    
    const wrapper = mount(<Todo text='test'/>);
  
    // wrapper.setState({ isEdit: true });
    expect(wrapper.state()).toEqual(
      { isEdit: false, text:'test' }
    );
  
    // here we use the "instance" method 
    // to get access to all instance methods defined on the component
    wrapper
      .instance()
      .showEdit();
  
      expect(wrapper.state()).toEqual(
        { isEdit: true, text:'test' }
      );
  });


  //check form
  it("update todo ", function() {
    
    const wrapper = mount(<Todo text='test'/>);
  
    wrapper.setState({ isEdit: true });
    // expect(wrapper.state()).toEqual(
    //   { isEdit: true, text:'test' }
    // );
  
    // here we use the "instance" method 
    // to get access to all instance methods defined on the component
    wrapper
      .instance()
      .showEdit();
    console.log('wrapper is',wrapper.debug())
    expect(wrapper.find('form').exists()).toEqual(true);
   
  });
