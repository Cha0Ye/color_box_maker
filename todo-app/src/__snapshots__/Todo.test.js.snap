// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Todo>
  <div
    className="Todo"
  >
    <div>
      <button>
        X
      </button>
      <button
        onClick={[Function]}
      >
         edit 
      </button>
    </div>
  </div>
</Todo>
`;
